packages/app/src/components/DBTraceWaterfallChart.tsx

```diff
@@ -22,233 +27,338 @@ type SpanRow = {
+      type,
+    }));
   }, [afterSpanData, beforeSpanData]);
 
-  const isSearchResultsFetching = isBeforeSpanFetching || isAfterSpanFetching;
-
   return {
-    data,
-    isFetching: isSearchResultsFetching,
-    aliasMap,
+    rows,
+    isFetching,
   };
 }
 
 // TODO: Optimize with ts lookup tables
 export function DBTraceWaterfallChartContainer({
   traceTableModel,
+  logTableModel,
   traceId,
   dateRange,
   focusDate,
   onClick,
   highlightedRowWhere,
 }: {
   traceTableModel: TSource;
+  logTableModel?: TSource;
   traceId: string;
   dateRange: [Date, Date];
   focusDate: Date;
-  onClick?: (rowWhere: string) => void;
+  onClick?: (rowWhere: { id: string; type: string }) => void;
   highlightedRowWhere?: string | null;
 }) {
-  const { data, isFetching, aliasMap } = useSpansAroundFocus({
-    traceTableModel,
+  const { rows: traceRowsData, isFetching: traceIsFetching } =
+    useSpansAroundFocus({
+      tableModel: traceTableModel,
+      focusDate,
+      dateRange,
+      traceId,
+    });
+  const { rows: logRowsData, isFetching: logIsFetching } = useSpansAroundFocus({
+    // search data if logTableModel exist
+    // search invliad date range if no logTableModel(react hook need execute no matter what)
+    tableModel: logTableModel || traceTableModel,
     focusDate,
-    dateRange,
+    dateRange: logTableModel ? dateRange : [dateRange[1], dateRange[0]],
     traceId,
   });
 
-  const rowWhere = useRowWhere({ meta: data?.meta, aliasMap });
-
-  const rows = useMemo(
-    () =>
-      data?.meta != null && data.meta.length > 0 // Sometimes meta has not loaded yet
-        ? data?.data?.map(row => {
-            return {
-              ...row,
-              id: rowWhere(omit(row, ['HyperDXEventType'])),
-            };
-          })
-        : undefined,
-    [data, rowWhere],
-  );
+  const isFetching = traceIsFetching || logIsFetching;
+  const rows = [...traceRowsData, ...logRowsData];
+
+  rows.sort((a, b) => {
+    const aDate = TimestampNano.fromString(a.Timestamp);
+    const bDate = TimestampNano.fromString(b.Timestamp);
+    const secDiff = aDate.getTimeT() - bDate.getTimeT();
+    if (secDiff === 0) {
+      return aDate.getNano() - bDate.getNano();
+    } else {
+      return secDiff;
+    }
+  });
 
   // 3 Edge-cases
   // 1. No spans, just logs (ex. sampling)
   // 2. Spans, but with missing spans inbetween (ex. missing intermediary spans)
   // 3. Spans, with multiple root nodes (ex. somehow disjoint traces fe/be)
 
-  const spanIds = useMemo(() => {
-    return new Set(
-      rows
-        ?.filter(result => result.HyperDXEventType === 'span')
-        .map(result => result.SpanId) ?? [],
-    );
-  }, [rows]);
-
   // Parse out a DAG of spans
-  type Node = SpanRow & { id: string; children: SpanRow[] };
+  type Node = SpanRow & { id: string; parentId: string; children: SpanRow[] };
   const rootNodes: Node[] = [];
-  const nodes: { [SpanId: string]: any } = {};
+  const parentSpanIdsMap = new Map();
```

Comment:
it's not clear to me what this map is being used for, on line 333 could we not pull the parent span id from the row itself?